> Cannot find module 'axios' from 'src/platform/service/k8s.js'

Axios is not available in shell-ui. I think we can simply use fetch as it is available in the browser we target

--- a/shell-ui/src/alerts/services/alertHooks.js
+++ b/shell-ui/src/alerts/services/alertHooks.js
@@ -27,10 +27,14 @@ export const getServicesAlertName = (): FilterLabels => {
 
 /**
  *
- * @param {FilterLabels} filterss
+ * @param {useContext} useContext
+ * @param {FilterLabels} filters
  * @returns An array of alerts with the highest severity
  */
-export const useHighestSeverityAlerts = (filters: FilterLabels): Alert[] => {
+export const useHighestSeverityAlerts = (
+  useContext: typeof useContext,
+  filters: FilterLabels,
+): Alert[] => {
   const query = useAlerts(useContext)(filters);
   const filteredAlerts = query && query.alerts;
 

--- a/shell-ui/src/alerts/services/alertHooks.test.js
+++ b/shell-ui/src/alerts/services/alertHooks.test.js
@@ -67,8 +67,8 @@ describe('useHighestSeverityAlerts hook', () => {
   afterEach(() => server.resetHandlers());
 
   const alertLibrary = window.shellUIAlerts[version];
-  console.log('window.shellUIAlerts', window.shellUIAlerts);
   alertLibrary.createAlertContext(createContext);
+
   const wrapper = ({ children }) => (
     <QueryClientProvider client={new QueryClient()}>
       <AlertProvider alertManagerUrl={testService}>{children}</AlertProvider>
@@ -80,7 +80,7 @@ describe('useHighestSeverityAlerts hook', () => {
   it('should only get the VolumeAtRisk alert when both VolumeAtRisk and VolumeDegraded are active', async () => {
     const { result, waitForNextUpdate } = renderHook(
       () =>
-        useHighestSeverityAlerts({
+        useHighestSeverityAlerts(useContext, {
           alertname: ['VolumeAtRisk', 'VolumeDegraded'],
         }),
       { wrapper },
@@ -95,7 +95,7 @@ describe('useHighestSeverityAlerts hook', () => {
   it('should get empty array', async () => {
     const { result, waitForNextUpdate } = renderHook(
       () =>
-        useHighestSeverityAlerts({
+        useHighestSeverityAlerts(useContext, {
           alertname: ['NodeAtRisk', 'NodeDegraded'],
         }),
       { wrapper },

--- a/shell-ui/src/platform/service/k8s.js
+++ b/shell-ui/src/platform/service/k8s.js
@@ -1,7 +1,6 @@
 //@flow
 import { useState, useEffect } from 'react';
 import { useQuery } from 'react-query';
-import axios from 'axios';
 import { AUTHENTICATED_EVENT } from '../../navbar/events';
 
 const useGetNodesCount = (useQuery: typeof useQuery, k8sUrl: string) => {
@@ -64,40 +63,13 @@ export const getNodesCountQuery = (
   k8sUrl: string,
   token?: string | null,
 ): typeof useQuery => {
-  if (!token) {
-    console.error('K8s API Not authenticated');
-    return;
-  }
   return {
-    cacheKey: 'countNodes',
+    queryKey: 'countNodes',
     queryFn: () =>
-      fetch(`${k8sUrl}/api/v1/nodes`)
-        .then((r) => {
-          if (r.ok) {
-            return r.json();
-          }
-        })
-        .then((res) => {
-          return res.items.length;
-        }),
-    options: { refetchInterval: 10000 },
-  };
-};
-
-export const getVolumesCountQuery = (
-  k8sUrl: string,
-  token?: string | null,
-): typeof useQuery => {
-  if (!token) {
-    console.error('K8s API Not authenticated');
-    return;
-  }
-  return {
-    cacheKey: 'countVolumes',
-    queryFn: () =>
-      fetch(`${k8sUrl}/api/v1/persistentvolumes`, {
+      fetch(`${k8sUrl}/api/v1/nodes`, {
         headers: {
-          authorisation: `bearer ${token}`,
+          // $FlowFixMe - if token is null, the query will not perform
+          authorization: `bearer ${token}`,
         },
       })
         .then((r) => {
@@ -108,6 +80,33 @@ export const getVolumesCountQuery = (
         .then((res) => {
           return res.items.length;
         }),
-    options: { refetchInterval: 10000 },
+    refetchInterval: 10000,
+    enabled: token ? true : false,
+  };
+};
+
+export const getVolumesCountQuery = (
+  k8sUrl: string,
+  token?: string | null,
+): typeof useQuery => {
+  return {
+    queryKey: 'countVolumes',
+    queryFn: () =>
+      fetch(`${k8sUrl}/api/v1/persistentvolumes`, {
+        headers: {
+          // $FlowFixMe - if token is null, the query will not perform
+          authorization: `bearer ${token}`,
+        },
+      })
+        .then((r) => {
+          if (r.ok) {
+            return r.json();
+          }
+        })
+        .then((res) => {
+          return res.items.length;
+        }),
+    refetchInterval: 10000,
+    enabled: token ? true : false,
   };
 };

